### PR TITLE
Added sample actor written in Go

### PR DESCRIPTION
**Pull request by artmello**
- Added actor definition files
- Added actor source code written in Go
- Added Makefile for actor building
- Added support for "build-cmd" into actor definition

### CODE REVIEW

**Comment by artmello** [comment]
@vinzenz / @shaded-enmity on Friday I spent sometime written a sample actor written in Go. I tried to use the approach that was discussed during last Cabal meeting. There is some questions that I don't know if I took the right approach (i.e. building actor before execution or adding a "build" method to snactor to be called by runner; passing building command via actor definition or always use "make" to build actor), but i am pushing this PR to share that.

**Review by vinzenz — CHANGES_REQUESTED** [review]
Please do not add the build step into the library - This should be done separately - e.g. during RPM build time  when - Or you have to do it manually or at least kick it off manually in the repo

File: examples/actors/goactor/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+BASE     = $(GOPATH)/src/$(PACKAGE)
+
+GO = go
+
+$(BASE):
+	@mkdir -p $(dir $@)
+	@ln -sf $(CURDIR) $@
+
+.PHONY: all
+all: | $(BASE) ; $(info $(M) building executable…)
```
Comment (vinzenz) [inline examples/actors/goactor/Makefile:12]: please add a build target for this

File: examples/actors/goactor/_actor.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+outputs:
+- name: sampleout
+  type:
+    name: SampleOut
+description: |
+  Sample actor written in Go
+execute:
+  executable: ./bin/goactor
+  build-cmd: make all
```
Comment (vinzenz) [inline examples/actors/goactor/_actor.yaml:9]: please do not specify this here

File: snactor/definition.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def __init__(self, name, init=None):
                 self.executable = os.path.abspath(os.path.join(self.base_path, self.executable))
             self.arguments = execute.get('arguments', [])
             self.output_processor = get_output_processor(execute.get('output-processor', None))
+            self.build_cmd = execute.get('build-cmd')
```
Comment (vinzenz) [inline snactor/definition.py:45]: ofc this has to be gone too

File: snactor/executors/default.py [hunk 4]
```diff
@@ -161,14 +161,34 @@ def execute(self, data):
             remote = self.definition.remote
             return self.execute_remote(data, resolve_variable_spec(data, remote.get('host', 'localhost')),
                                        resolve_variable_spec(data, remote.get('user', 'root')))
+
```
Comment (vinzenz) [inline snactor/executors/default.py:164]: and this section as well

**Comment by artmello** [comment]
After discussion under Cabal meeting it was clear that snactor should not be responsible for building the actor and changes to YAML file would be unnecessary. So, I will close this PR

**Pull request closed without merging**